src/config-validator.ts

```diff
@@ -0,0 +1,121 @@
+import {z, ZodArray, ZodRecord, ZodString} from "zod";
+import {SchemaInfo} from "./middlewares/datasource";
+
+
+function getRequiredEnvVar(name: string){
+    const property = process.env[name] as any;
+    if(!property || property.trim().length==0){
+        throw new Error("config file is missing property "+name);
+    }
+    return property;
+}
+
+
+export const getRequiredNumber = (name: string): number => {
+    const value = parseInt(getRequiredEnvVar(name));
+    const numberValue = z.number().finite().safeParse(value);
+    if(!numberValue.success) {
+        throw new Error("property "+name+" in config is not a valid number")
+    }
+    return numberValue.data;
+}
+
+
+export const getRequiredString = (name: string): string => {
+    const value = getRequiredEnvVar(name);
+    const stringValue = z.string().safeParse(value)
+    if(!stringValue.success) {
+        throw new Error("property "+name+" in config is not a valid string");
+    }
+    return stringValue.data;
+}
+
+
+export const getString = (name:string): string => {
+    const value = process.env[name] || '';
+    const stringValue = z.string().safeParse(value)
+    if(!stringValue.success) {
+        throw new Error("property "+name+" in config is not a valid string");
+    }
+    return stringValue.data;
+}
+
+
+export const getUrl = (name:string): string => {
+    const value = process.env[name] || 'a://';
+    const stringValue = z.string().url().safeParse(value)
+    if(!stringValue.success) {
+        throw new Error("property "+name+" in config is not a valid string");
+    }
+    return stringValue.data;
+}
+
+
+export const getRequiredArray = (name: string): ZodString["_output"][] => {
+    const value = JSON.parse(process.env[name] || '[]');
+    if(value.length == 0){
+        throw new Error("config file is missing property "+name);
+    }
+    return getArray(name);
+}
```

Comment:
This performs `JSON.parse` twice on the same data. Very easy to simplify:

```suggestion
export const getRequiredArray = (name: string): ZodString["_output"][] => {
    const value = getArray(name);
    if(value.length == 0){
        throw new Error(`Environment variable "${name}" is invalid. It must be an array of strings with at least one entry.");
    }
    return value;
}
```

Reply:
yes, that's much better. Thanks.